### PR TITLE
feat(provider/kubernetes): write up some best practices

### PR DESCRIPTION
**Pull request by lwander**

### CODE REVIEW

File: guides/user/kubernetes-v2/best-practices/index.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+deployed, it will point to the same contents (binary, dependencies, config,
+etc...). If you rely on the tag, it's entirely possible that deploying the same
+Manifest twice can have different results.
+
+There are two ways to achieve this in Spinnaker:
+
+1. Template your manifests outside of Spinnaker to point to the digest you want
+   to deploy, before submitting them to Spinnaker.
+
+2. Rely on Spinnaker's artifact substitution and a trigger that supplies your
```
Comment (dorbin) [inline guides/user/kubernetes-v2/best-practices/index.md]: Link here to how to do artifact substitution?

File: guides/user/kubernetes-v2/best-practices/index.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+The reasoning is: the digest is a content-addressable way of referencing
+your image, since it's derived from the hash of the image contents. If your
+manifest points to an image by its digest, you know that each time it's
+deployed, it will point to the same contents (binary, dependencies, config,
+etc...). If you rely on the tag, it's entirely possible that deploying the same
+Manifest twice can have different results.
+
+There are two ways to achieve this in Spinnaker:
+
+1. Template your manifests outside of Spinnaker to point to the digest you want
```
Comment (dorbin) [inline guides/user/kubernetes-v2/best-practices/index.md:30]: Maybe elaborate, for some set of less K8s-savvy users, what it means to template one's manifest. Or just link to something if there's anything good to link to.

File: guides/user/kubernetes-v2/best-practices/index.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+   image digest.
+
+## Emit Your Deployed Manifests to an Audit Log
+
+To keep track of what's changing in your cluster, and under what circumstances
+it is changing, we highly recommend configuring Spinnaker to emit events to an
+external audit log. Spinnaker will already generate events for any events
+running through its orchestration engine, all that's needed is an endpoint to
+send them to. [This guide shows you
+how](https://blog.spinnaker.io/spinnaker-echo-google-cloud-functions-stackdriver-logging-spinnaker-audit-log-81139f084db9).
```
Comment (dorbin) [inline guides/user/kubernetes-v2/best-practices/index.md]: You could maybe omit this sentence and instead link from "...configuring Spinnaker to emit events to an external audit log."

File: guides/user/kubernetes-v2/best-practices/index.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+## Version Your ConfigMaps and Secrets
+
+Anytime Spinnaker deploys a ConfigMap or Secret, it will append a version to
+its name. If that exact ConfigMap or Secret is already running in the cluster,
+it won't be redeployed. Downstream stages that reference that ConfigMap or
+Secret will deploy the version chosen by Spinnaker. __Unless your application
+requires hot-reloading of configuration, this is essential for practicing
+safe delivery__. Let's go through some examples:
+
+### 1. You Want to Slowly Rollout a Configuration Change
```
Comment (dorbin) [inline guides/user/kubernetes-v2/best-practices/index.md]: I wouldn't number these, unless for some reason they're ordered.

File: guides/user/kubernetes-v2/best-practices/index.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+Say you are doing a rollout of a new feature, hidden behind a feature-flag. All
+that's needed is an update to a ConfigMap referenced by a Deployment.
+
+If your Deployment has mounted this ConfigMap, pushing a change to it will
+immediately allow all Pods to read the update, rendering any sort of rollout
+useless. This will happen even if you
```
Comment (dorbin) [inline guides/user/kubernetes-v2/best-practices/index.md]: "This will happen even if you...fall asleep while writing the doc" ;)

**Review by dorbin — APPROVED** [review]

File: guides/user/kubernetes-v2/best-practices/index.md (hunk 5 above)
Comment (lwander) [reply guides/user/kubernetes-v2/best-practices/index.md]: heh :)

**Pull request merged**